src/blog/understanding-ecmascript-part-4.md

```diff
@@ -24,15 +24,33 @@ In [part 3](/blog/understanding-ecmascript-part-3), we familiarized ourselves wi
 
 ## Cover grammars
 
-In this episode, we take a deeper look into *cover grammars*. They are a way to specify grammar rules for syntactic constructs which look ambiguous at first.
+In this episode, we take a deeper look into *cover grammars*. They are a way to specify the grammar for syntactic constructs which look ambiguous at first.
 
 Again, we'll skip the subscripts for `[In, Yield, Await]` for brevity, as they aren't important for this blog post. See [part 3](/blog/understanding-ecmascript-part-3) for an explanation of their meaning and usage.
 
-## Parenthesized expression or an arrow parameter list?
+## Finite lookaheads
 
-Typically, parsers decide which grammar production to follow based on finite lookahead (a fixed amount of following tokens).
+Typically, parsers decide which production to use based on a finite lookahead (a fixed amount of following tokens).
+
+In some cases, the next token determines the production to use unambiguously. [For example](https://tc39.es/ecma262/#prod-UpdateExpression):
+
+```grammar
+UpdateExpression :
+  LeftHandSideExpression
+  LeftHandSideExpression ++
+  LeftHandSideExpression --
+  ++ UnaryExpression
+  -- UnaryExpression
+```
+
+If we're parsing an `UpdateExpression` and the next token is `++` or `--`, we know the production to use right away. If the next token is neither, it's still not too bad: we can parse a `LeftHandSideExpression` starting from the position we're at, and figure out what to do after we've parsed it. If the token following the `LeftHandSideExpression` is `++`, the production to use is `UpdateExpression : LeftHandSideExpression ++`. The case for `--` is similar. And if the token following the `LeftHandSideExpression` is neither `++` nor `--`, we use the production `UpdateExpression : LeftHandSideExpression`.
```

Comment:
Maybe a parenthetical saying that for the those familiar with formal grammars, the grammar is LR(1)? Up to you.

Reply:
I was deliberately leaving it out, afaics it would be inaccurate to say it's LR(1), it's rather "LR(1) with some reasonable extensions" and I don't want to go in the details there.